xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/script/NotificationPreferenceScriptService.java

```diff
@@ -292,7 +297,26 @@ public NotificationEmailDiffType getDiffType()
      */
     public NotificationEmailDiffType getDiffType(String userId)
     {
-        return emailUserPreferenceManager.getDiffType(userId);
+        return emailUserPreferenceManager.getDiffType(userReferenceResolver.resolve(userId));
+    }
+
+    /**
+     * @return the email notification interval configured for the current user
+     * @since 14.4.2
+     */
+    public NotificationEmailInterval getInterval()
+    {
+        return emailUserPreferenceManager.getInterval();
+    }
+
+    /**
+     * @param userId id of a user
+     * @return the notification email interval configured for the given user
+     * @since 14.4.2
+     */
+    public NotificationEmailInterval getInterval(String userId)
```

Comment:
Normally you can directly use a `UserReference` as parameter there should be an automatic conversion String -> UserReference.

Reply:
thx done